### DISCUSSION TITLE
Is it possible to match wildcard URLs?

### DISCUSSION THREAD

**Discussion by ghost** [ory/keto#857] [Q&A] [answered]
The Object documentation states, "It's however possible to use for example URLs or opaque tokens of any kind. Please check the limitations. Ory Keto will consider objects equal iff their string representation is equal." - [doc](https://www.ory.sh/docs/keto/concepts/objects)

This implies to me that:
"example.com/share/cats/video.mp4" != "example.com/share/**"
and thus does not match.

Is there glob or regex or custom matching with which this could be accomplished? I wasn't able to find any other mentions to how objects are matched.

**Comment by vinckr** [comment]
I think the wildcard should work though. So if the string representation is equal = if the wildcard substitution is equal. 

See also these comments: https://github.com/ory/keto/issues/608#issuecomment-861460156

I am not 100% sure if it works as I expect it to, so pinging @zepatrik for quick comment and we can amend the documentation accordingly :-)

**Comment by zepatrik** [comment, marked as answer]
No, there is no support for wildcard matching. As the documentation says, the strings have to be exactly equal.
What is your use-case? If you want to apply the relations of the parent to the children, you should rather track the parent-child relationship through relation tuples. There is an example that shows that, but it is not very complete: https://www.ory.sh/docs/keto/guides/expand-api-display-who-has-access
Also, currently you will have to work a bit around #263 as explained in the example.

**Comment by ghost** [comment]
Thank you for the replies :)

@zepatrik
As far as I can tell all of the examples have a 1 to 1 match of object and relationship, and I'm wondering if it's possible to avoid defining relationships for subdirectories which I don't want to specify (yet?)
For example:
```
├─ example.com                  (owner: maureen)
   ├─ personal                  (owner: maureen)
   ├─ share                     (owner: maureen)
      ├─ share-with-laura       (owner: maureen; shared with laura)
         ├─ one
            ├─ subone
            ├─ subtwo
         ├─ two
         ├─ ...
         ├─ ten-million
```
In this example, maureen wants to share "share-with-laura" with laura for all the subfolders, but there's a lot of them (ten-million!), and maybe they all have subfolders. Do I have to define all these relationships?

(I only showed the folder structure in my example, but the objects would be something like:
`example.com/share` is parent of `example.com/share/share-with-laura`
or I believe something similar to:
`example.com/share#parent@(example.com/share/share-with-laura#_)`
)

What I would like to do is basically:
"Share `example.com/share/share-with-laura/**` with `laura`"
This way I could ask:
"Does `laura` have access to `example.com/share/share-with-laura/one/subone/subsubone/.../file.ext`? Yes!"

**Reply by zepatrik** [reply]
If you track the parent relationship during object creation/update/deletion, you will already have all those tuples in the system. Then you would use #263 to define that read on a parent induces read on the child (not yet possible through config). Lastly, when sharing the parent with someone, the only tuple you have to add is `parent#read@user`. Through the tracked parent relationship and the subject-set-rewrite the user will automatically inherit the read relation on all children.
